packages/backend-auth/package.json

```diff
@@ -10,6 +10,11 @@
       "types": "./lib/index.d.ts",
       "import": "./lib/index.js",
       "require": "./lib/index.js"
+    },
+    "./internal/iam_actions": {
+      "types": "./lib/internal/iam_actions.d.ts",
+      "import": "./lib/internal/iam_actions.js",
+      "require": "./lib/internal/iam_actions.js"
```

Comment:
will this show up in IDE syntax suggestions?

Reply:
You mean like this? or from the amplify DX? 
<img width="677" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/9db51795-1c0d-4d99-aca6-7251bc455180">

Reply:
> or from the amplify DX?

From amplify DX. Are customers going to see possibility to import these from `@aws-amplify/backend-auth/internal` ?

Reply:
Ok, so if the question is about doing this:
<img width="1997" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/110861985/70dc12e1-62a2-4cb9-99cb-ed54772b83f3">

There is no intellisense and you must manually specify the /internal/iam_actions path to be able to import it

Reply:
https://github.com/aws-amplify/amplify-backend/assets/110861985/9ca997df-d782-4170-8ba7-4b96fe95045e

Reply:
I get this 
<img width="975" alt="image" src="https://github.com/aws-amplify/amplify-backend/assets/5849952/57f1a77e-7fcc-46d2-aae5-ca3b5e54bb1a">


When trying to use similar export from here https://github.com/aws-amplify/amplify-backend/blob/4dd9d5a35c378895d1360c15a3b7ad1f09cc7653/packages/client-config/package.json#L14-L18

Reply:
I'm okay with this as-is, but if we want an even stronger suggestion that customers shouldn't use this, we could namespace it `internal-only-do-not-use/iam-actions` or something to that effect

Reply:
If you want to try the same DX; try importing from  '@aws-amplify/backend/types/platform'

import * as types from '@aws-amplify/backend/types/platform';

Reply:
To be clear. The goal is to make it as hidden as possible.

We may say that `<root>/internal/*` is standard way of saying "public but don't use it". However, we haven't completed https://github.com/aws-amplify/amplify-backend/issues/1533 . So either we should define that first or treat this PR as precedent.